packages/combo-box/src/vaadin-combo-box-mixin.js

```diff
@@ -536,7 +536,11 @@ export const ComboBoxMixin = (subclass) =>
         // For touch devices, we don't want to popup virtual keyboard
         // unless input element is explicitly focused by the user.
         if (!this._isInputFocused() && !isTouch) {
-          this.focus();
+          if (this.inputElement) {
+            this._closeOnBlurIsPrevented = true;
```

Comment:
I used `_closeOnBlurIsPrevented` here to keep backward compatibility. Without it, the following two tests are failing

```
custom buttons > toggle-button > should open overlay on toggle button Arrow Down
custom buttons > toggle-button > should open overlay on toggle button Arrow Up
```

But maybe there is a better way how to solve this...

Reply:
This logic is there due to historical reasons and is intended to support focusable custom toggle button.

We might want to refactor related logic later. But for now, I think the solution is fine.

Reply:
Have you considered fixing this in `_onFocusout()`? It looks like an appropriate place to prevent closing when focus is moving within the combo-box.

Reply:
Yes, I agree that `_onFocusout()` is the right place, I just couldn't come up with a fix without digging deeper.

Reply:
Good points, thanks! In the end I've decided to:
1. fix it in `_onFocusout()`
2. but do it in the `vaadin-combo-box-light` only by overriding `_onFocusout()` there
3. Make the behavior change very isolated by affecting the `focusOut` really only when moving the focus from the toggle/clear button to the internal input - 

That way, I believe, the impact on focus behavior is minimal and hopefully, my changes have a minimal impact on existing focus behavior